Dependency cert-manager missing from v0.1.14 on

I recently tried out the new release v0.1.14 and ran into the problem that make deploy would fail with the following error message.

```
$ make deploy IMG=ghcr.io/spotify/flink-operator:v0.1.14
[...]
service/flink-operator-webhook-service created
deployment.apps/flink-operator-controller-manager created
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
mutatingwebhookconfiguration.admissionregistration.k8s.io/flink-operator-mutating-webhook-configuration created
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
validatingwebhookconfiguration.admissionregistration.k8s.io/flink-operator-validating-webhook-configuration created
Error from server (NotFound): error when creating "STDIN": the server could not find the requested resource (post certificates.cert-manager.io)
Error from server (NotFound): error when creating "STDIN": the server could not find the requested resource (post issuers.cert-manager.io)
make: *** [deploy] Error 1
```

The problem is that the CRDs for `Certificate` and `Issuer` are missing. I found out that they are provided by the cert-manager project.

Installing cert-manager fixes the problem, but I wanted to report this, because I think it would be nice to have this documented or fixed in the build logic.

```
kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.5.3/cert-manager.yaml
```

If you guide me I could try to provide a pull request.

--- a/README.md
+++ b/README.md
@@ -15,9 +15,9 @@ The operator is under active development, backward compatibility of the APIs is 
 
 ## Prerequisites
 
-- Version >= 1.15 of Kubernetes
-- Version >= 1.15 of kubectl (with kustomize)
-- Version >= 1.7 of Apache Flink
+- Version >= 1.15 of [Kubernetes](https://kubernetes.io)
+- Version >= 1.7 of [Apache Flink](https://flink.apache.org)
+- Version >= 1.5.3 of [cert-manager](https://cert-manager.io)
 
 ## Overview
 

--- a/docs/developer_guide.md
+++ b/docs/developer_guide.md
@@ -28,19 +28,16 @@ You might want to watch this tech talk ([video](https://www.youtube.com/watch?v=
 The following dependencies are required to build the Flink Operator binary and
 run unit tests:
 
-* [Go v1.14+](https://golang.org/)
-* [Kubebuilder v2+](https://github.com/kubernetes-sigs/kubebuilder)
+- [Go v1.14+](https://golang.org/)
+- [Kubebuilder v3+](https://github.com/kubernetes-sigs/kubebuilder)
 
-But you don't have to install them on your local machine, because this project
-includes a [builder Docker image](../Dockerfile.builder) with the dependencies
-installed. Build and unit test can happen inside of the builder container. This
-is the recommended way for local development.
+But you don't have to install them on your local machine, the [`Makefile`](../Makefile) will take care of those dependencies.
 
-But to create the Flink Operator Docker image and deploy it to a Kubernetes
+To create the Flink Operator Docker image and deploy it to a Kubernetes
 cluster, the following dependencies are required on your local machine:
 
-* [Docker](https://docs.docker.com/install/)
-* [kubectl v1.15+](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [Docker](https://docs.docker.com/install/)
+- [kubectl v1.15+](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
 
 ## Local build and test
 
@@ -56,7 +53,7 @@ Build a Docker image for the Flink Operator and then push it to an image
 registry with
 
 ```bash
-make operator-image push-operator-image IMG=<tag>
+make docker-push IMG=<tag>
 ```
 
 For example, if you are using [Google Container Registry](https://cloud.google.com/container-registry/docs/),
@@ -66,12 +63,8 @@ to set it up, then build the image and push it to GCR with:
 ```bash
 PROJECT=<gcp-project>
 IMAGE_TAG=gcr.io/${PROJECT}/flink-operator:latest
-make operator-image push-operator-image IMG=${IMAGE_TAG}
+make docker-push IMG=${IMAGE_TAG}
 ```
-
-After building the image, the image tag will be automatically saved in
-`config/default/manager_image_patch.yaml`, when you run `make deploy` later,
-it knows what image to use.
 
 ## Deploy the operator and run jobs
 

--- a/docs/user_guide.md
+++ b/docs/user_guide.md
@@ -5,50 +5,51 @@ a Kubernetes cluster and running a sample Flink job.
 
 ## Prerequisites
 
-* Get a running Kubernetes cluster, you can verify the cluster info with
+- Get a running Kubernetes cluster, you can verify the cluster info with
 
   ```bash
   kubectl cluster-info
   ```
 
-* Clone the Flink Operator repo to your local machine:
+- Install cert-manager
 
   ```bash
-  git clone git@github.com:GoogleCloudPlatform/flink-on-k8s-operator.git
+  kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.5.3/cert-manager.yaml
+  ```
+
+- (Optional) Clone the Flink Operator repo to your local machine:
+  (This step is not needed if you choose to install through kubectl or Helm Chart).
+
+  ```bash
+  git clone git@github.com:spotify/flink-on-k8s-operator.git
   ```
 
   then switch to the repo directory, we need to use the scripts in the repo for
-  deployment. (This step is not needed if you choose to install through Helm Chart).
+  deployment.
 
-  To execute the install scripts, you'll also need [kubectl v1.14+](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
-  and [go v1.12+](https://golang.org/doc/install) installed on your local machine.
-
-* (Optional) Choose a Flink Operator image.
-
-  By default, the deployment uses the image `ghcr.io/spotify/flink-operator:latest`,
-  but you can find and choose other released images by running the following command:
-
-  ```bash
-  gcloud container images list-tags gcr.io/flink-operator/flink-operator
-  ```
-
-  You can also follow the [Deverloper Guide](./developer_guide.md) to build your
-  own image from the source code and push the image to a registry where your
-  Kubernetes cluster can access.
+  To execute the install scripts, you'll need [go v1.14+](https://golang.org/doc/install) installed on your local machine.
 
 ## Deploy the operator to a Kubernetes cluster
 
 You can deploy the Flink Operator to the Kubernetes cluster through one of the
-following 2 ways:
+following 3 ways:
 
-* **Option 1: Make deploy**
+- **Option 1: Using `kubectl`**
+
+  Pick a version at [https://github.com/spotify/flink-on-k8s-operator/releases](https://github.com/spotify/flink-on-k8s-operator/releases) and
+
+  ```bash
+  kubectl apply -f https://github.com/spotify/flink-on-k8s-operator/releases/download/$VERSION/flink-operator.yaml
+  ```
+
+- **Option 2: `make deploy`**
 
   Simply run
-  
+
   ```base
   make deploy
   ```
-  
+
   from the source repo to deploy the operator. There are some flags which you
   can use to configure the deployment:
 
@@ -60,24 +61,24 @@ following 2 ways:
       [WATCH_NAMESPACE=<namespace-to-watch>]
   ```
 
-  * `IMG`: The Flink Operator image. The default value is `ghcr.io/spotify/flink-operator:latest`.
-  * `FLINK_OPERATOR_NAMESPACE`: the namespace of the operator. The default value is
+  - `IMG`: The Flink Operator image. The default value is `ghcr.io/spotify/flink-operator:latest`.
+  - `FLINK_OPERATOR_NAMESPACE`: the namespace of the operator. The default value is
     `flink-operator-system`.
-  * `RESOURCE_PREFIX`: the prefix to avoid conflict of cluster-scoped resources.
+  - `RESOURCE_PREFIX`: the prefix to avoid conflict of cluster-scoped resources.
     The default value is `flink-operator-`.
-  * `WATCH_NAMESPACE`: the namespace of the `FlinkCluster` CRs which the operator
+  - `WATCH_NAMESPACE`: the namespace of the `FlinkCluster` CRs which the operator
     watches. The default value is empty string which means all namespaces.
-  
+
   **Note:** It is highly recommended to just use the default values unless you want to
   deploy multiple instances of the operator in a cluster, see more details in the
   How-to section of this doc.
 
-* **Option 2: Helm Chart**
+- **Option 3: Helm Chart**
 
   Follow the [Helm Chart Installation Guide](../helm-chart/flink-operator/README.md) to
   install the operator through Helm Chart.
 
-Note, for kubernetes cluster with private cluster domain, you should add a CLUSTER\_DOMAIN environment to the operator deployment.
+Note, for kubernetes cluster with private cluster domain, you should add a CLUSTER_DOMAIN environment to the operator deployment.
 
 ## Verify the deployment
 
@@ -172,12 +173,12 @@ Pod to terminate)
 
 There are several ways to submit jobs to a session cluster.
 
-* **Flink web UI**
+- **Flink web UI**
 
   You can submit jobs through the Flink web UI. See instructions in the
   Monitoring section on how to setup a proxy to the Flink web UI.
 
-* **From within the cluster**
+- **From within the cluster**
 
   You can submit jobs through a client Pod in the same cluster, for example:
 
@@ -205,7 +206,7 @@ There are several ways to submit jobs to a session cluster.
   EOF
   ```
 
-* **From outside the cluster**
+- **From outside the cluster**
 
   If you have configured the access scope of JobManager as `External` or `VPC`,
   you can submit jobs from a machine which is in the scope, for example:
@@ -408,29 +409,32 @@ See this [doc](./savepoints_guide.md) on how to manage savepoints with the opera
 ### Update Flink clusters and jobs
 
 To update a running Flink job's program or execution settings, you can create new savepoint, terminate the job,
-and create a new FlinkCluster with the new program, settings, and savepoint. 
+and create a new FlinkCluster with the new program, settings, and savepoint.
 However, in some cases, you may want to update the Flink job while maintaining the logical continuity
 of the Flink job with the FlinkCluster custom resource. In this case, you can continuously update
 the FlinkCluster custom resource, and the Flink operator takes care of the process required to update
 the Flink job and cluster.
 
 There are several points to note when using the job update feature.
-* To use the job update feature, `savepointsDir` must be set and the value of this field cannot be deleted when updating.
-This is because the Flink operator requires it to create a savepoint for job updates.
-* You can resume Flink job from your desired savepoint by updating `fromSavepoint`.
-If you want to resume the updated job from the latest savepoint, `fromSavepoint` must be unspecified.
-* `cancelRequested` and `savepointGeneration` are not allowed to update at the same time with other fields
-due to functional characteristics.
+
+- To use the job update feature, `savepointsDir` must be set and the value of this field cannot be deleted when updating.
+  This is because the Flink operator requires it to create a savepoint for job updates.
+- You can resume Flink job from your desired savepoint by updating `fromSavepoint`.
+  If you want to resume the updated job from the latest savepoint, `fromSavepoint` must be unspecified.
+- `cancelRequested` and `savepointGeneration` are not allowed to update at the same time with other fields
+  due to functional characteristics.
 
 There are some behavioral characteristics in update.
-* Whenever the FlinkCluster spec is updated, the Flink operator creates
-a [ControllerRevision](https://godoc.org/k8s.io/api/apps/v1#ControllerRevision) resource
-that stores the changed spec. ControllerRevisions can be used to check the editing history.
-* If update is triggered while the cluster is running, all components are re-created after terminated.
-If update is triggered in terminated state, all components are re-created as well.
-* When job is to be updated, the Flink operator will restore the job from the latest savepoint available
-- `savepointLocation` or `fromSavepoint` in job status.
-If those are not available, the job is restarted from the beginning.
+
+- Whenever the FlinkCluster spec is updated, the Flink operator creates
+  a [ControllerRevision](https://godoc.org/k8s.io/api/apps/v1#ControllerRevision) resource
+  that stores the changed spec. ControllerRevisions can be used to check the editing history.
+- If update is triggered while the cluster is running, all components are re-created after terminated.
+  If update is triggered in terminated state, all components are re-created as well.
+- When job is to be updated, the Flink operator will restore the job from the latest savepoint available
+
+* `savepointLocation` or `fromSavepoint` in job status.
+  If those are not available, the job is restarted from the beginning.
 
 For example, you can create [wordcount job v1.9.2](../examples/update/wordcount-1.9.2.yaml)
 and update it to [wordcount job v1.9.3](../examples/update/wordcount-1.9.3.yaml) like this.
@@ -451,29 +455,31 @@ kubectl get controllerrevision <REVISION-NAME> -o yaml
 ### Control Logging Behavior
 
 The default logging configuration provided by the operator sends logs from JobManager and TaskManager to `stdout`. This
-has the effect of making it so that logging from Flink workloads running on Kubernetes behaves like every other 
-Kubernetes pod. Your Flink logs should be stored wherever you generally expect to see your container logs in your 
+has the effect of making it so that logging from Flink workloads running on Kubernetes behaves like every other
+Kubernetes pod. Your Flink logs should be stored wherever you generally expect to see your container logs in your
 environment.
 
-Sometimes, however, this is not a good fit. An example of when you might want to customize logging behavior is to 
-restore the visibility of logs in the Flink JobManager web interface. Or you might want to ship logs directly to a 
-different sink, or using a different formatter. 
+Sometimes, however, this is not a good fit. An example of when you might want to customize logging behavior is to
+restore the visibility of logs in the Flink JobManager web interface. Or you might want to ship logs directly to a
+different sink, or using a different formatter.
 
 You can use the `spec.logConfig` field to fully control the log4j and logback configuration. It is a string-to-string map,
-whose keys and values become filenames and contents (respectively) in the folder `/opt/flink/conf` in each container. 
-The default Flink docker entrypoint expects this directory to contain two files: `log4j-console.properties` and 
+whose keys and values become filenames and contents (respectively) in the folder `/opt/flink/conf` in each container.
+The default Flink docker entrypoint expects this directory to contain two files: `log4j-console.properties` and
 `logback-console.xml`.
 
-An example of using this parameter to make logs visible in both the Flink UI and on stdout 
+An example of using this parameter to make logs visible in both the Flink UI and on stdout
 [can be found here](../examples/log_config.yaml).
 
 ### Control Security and Permissions in Pods
-You can set various security-related attributes of the JobManager, TaskManager, and Job Pods using a 
+
+You can set various security-related attributes of the JobManager, TaskManager, and Job Pods using a
 [PodSecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podsecuritycontext-v1-core)
 object. It is possible to run the entrypoint of the container process as a different user or group,
-and to modify ownership of mounted volumes. 
+and to modify ownership of mounted volumes.
 
-You can set the SecurityContext in the FlinkCluster spec, within the JobManager, TaskManager, and Job fields, like this: 
+You can set the SecurityContext in the FlinkCluster spec, within the JobManager, TaskManager, and Job fields, like this:
+
 ```yaml
 taskManager:
   ...
@@ -482,12 +488,13 @@ taskManager:
     runAsGroup: 1000
     fsGroup: 2000
 ```
+
 You can set different SecurityContexts for the TaskManager, JobManager StatefulSets and the Job, but all TaskManager pods
 will share the same one.
 Examples and explanations of the available options can be found [here](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod).
 
 ### Mounting external volumes to the pods
+
 If your deployment requires larger storage captivity, or a faster access to the state backend you can use `volumeClaimTemplates` option in TaskManager config
 to create a new claim template and then mount it in `volumeMounts`  
 Check the [FlinkCluster Custom Resource Definition](./crd.md) and [StatefulSet's doc](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) for more info
-

